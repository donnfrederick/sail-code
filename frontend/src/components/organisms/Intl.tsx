import React, {
  PropsWithChildren,
  Dispatch,
  SetStateAction,
  useContext,
  useState
} from 'react'
import { IntlProvider } from 'react-intl'
import { messages, browserLocale, Locale } from 'utils/intl'

export type LocaleState = {
  locale: Locale
}

const initialState: LocaleState = {
  locale: browserLocale
}

const IntlStateContext = React.createContext<LocaleState>(initialState)

const SetLocaleStateContext = React.createContext<
  Dispatch<SetStateAction<LocaleState>>
>(() => {
  // empty block
})

export function useLocaleState() {
  return useContext(IntlStateContext)
}
export function useSetLocaleState() {
  return useContext(SetLocaleStateContext)
}

export function LocaleStateProvider(props: {
  initialState?: LocaleState
  children: React.ReactNode
}) {
  const [state, setState] = useState<LocaleState>(
    props.initialState || initialState
  )
  return (
    <IntlStateContext.Provider value={state}>
      <SetLocaleStateContext.Provider value={setState}>
        {props.children}
      </SetLocaleStateContext.Provider>
    </IntlStateContext.Provider>
  )
}

export const Intl = (props: PropsWithChildren<{}>) => {
  const localeState = useLocaleState()

  return (
    <LocaleStateProvider>
      <IntlProvider
        messages={messages[localeState.locale]}
        locale={localeState.locale}
        defaultLocale="en"
      >
        {props.children}
      </IntlProvider>
    </LocaleStateProvider>
  )
}
