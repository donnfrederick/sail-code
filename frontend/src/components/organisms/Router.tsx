import * as Sentry from '@sentry/browser'
import ErrorPage from 'components/pages/Error'
import { createBrowserHistory } from 'history'
import * as React from 'react'
import { Provider, Store } from 'react-redux'
import { ConnectedRouter } from 'react-router-redux'
import { compose, lifecycle, withState } from 'recompose'
import createStore from 'store/createStore'
import resetViewport from 'utils/resetViewport'
import userAgent from 'utils/userAgent'

declare const gtag: any

export const history = createBrowserHistory()
export const store = createStore(history) as Store<any>

history.listen(() => {
  window.scrollTo(0, 0)
  gtag('config', 'UA-105776977-2', { page_path: history.location.pathname })
  userAgent()
  resetViewport()
})

interface Props {
  children: JSX.Element
  errors: Error | null
  setErrors: (state: any) => any
}

const isProd = process.env.NODE_ENV === 'production'

const sentry = compose(
  withState('errors', 'setErrors', null),
  isProd
    ? lifecycle<Props, {}, {}>({
        componentDidCatch(error, errorInfo) {
          this.props.setErrors(error)
          Sentry.withScope(scope => {
            Object.keys(errorInfo).forEach(key => {
              scope.setExtra(key, errorInfo[key])
            })
            Sentry.captureException(error)
          })
        }
      })
    : lifecycle<Props, {}, {}>({
        componentDidCatch(error) {
          this.props.setErrors(error)
        }
      })
)

export default sentry((props: Props) => {
  const { children, errors } = props

  return (
    <Provider store={store}>
      <ConnectedRouter history={history}>
        {errors ? <ErrorPage /> : children}
      </ConnectedRouter>
    </Provider>
  )
})
