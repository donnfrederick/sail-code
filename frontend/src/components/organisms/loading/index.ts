import Loading from 'components/organisms/loading/Loading'
import { connect } from 'react-redux'
import { State as RootState } from 'reducers'

// tslint:disable-next-line
interface OuterProps {}

interface Props {
  isFetchingConversation: boolean
  isFetchingLocations: boolean
  isFetchingNotification: boolean
  isFetchingOrganizations: boolean
  isFetchingStudentsMe: boolean
  isFetchingTeachersMe: boolean
}

const mapStateToProps = (state: RootState, props: OuterProps): Props => {
  return {
    isFetchingConversation: state.rootReducer.conversation.fetching > 0,
    isFetchingLocations: state.rootReducer.locations.isFetching,
    isFetchingNotification: state.rootReducer.notification.isFetching,
    isFetchingOrganizations: state.rootReducer.organizations.isFetching,
    isFetchingStudentsMe: state.rootReducer.students.isFetching,
    isFetchingTeachersMe: state.rootReducer.teachers.isFetching
  }
}

const enhancer: any = connect(
  mapStateToProps,
  {}
)

export default enhancer(Loading)
