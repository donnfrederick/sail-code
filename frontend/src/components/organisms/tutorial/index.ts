import Tutorial from 'components/organisms/tutorial/Tutorial'
import { connect } from 'react-redux'
import { State as RootState } from 'reducers'
import * as TutorialActions from 'reducers/tutorial'

// tslint:disable-next-line
interface OuterProps {}

interface Props {
  shouldShow: boolean
  showConfirmation: boolean
}

const mapStateToProps = (state: RootState, props: OuterProps): Props => {
  return {
    shouldShow: state.rootReducer.tutorial.shouldShow,
    showConfirmation: state.rootReducer.teachers.showConfirmation
  }
}

const enhancer: any = connect(
  mapStateToProps,
  {
    hide: TutorialActions.hide,
    show: TutorialActions.show
  }
)

export default enhancer(Tutorial)
