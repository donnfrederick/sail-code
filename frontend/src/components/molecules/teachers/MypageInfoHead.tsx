import Week from 'components/atoms/teachers/Week'
import { store } from 'components/organisms/Router'
import * as ConversationModels from 'models/conversation'
import moment from 'moment'
import * as React from 'react'
import { getConversations } from 'reducers/conversation'
import styled from 'styled-components'
import isHuawei from 'utils/isHuawei'
import resolvePath from 'utils/resolvePath'

interface Props {
  authToken: string
  page: number
  reservations: ConversationModels.ConversationResponse | null
  back(): void
  forward(): void
}

export default (props: Props) => {
  const { authToken, back, forward, page, reservations } = props

  const startOn = reservations
    ? ''
    : moment()
        .startOf('week')
        .format('YYYY-MM-DD') + ''
  return (
    <Container>
      {reservations ? (
        <Week reservations={reservations} start={reservations.meta.start_on} />
      ) : (
        <Week reservations={null} start={startOn} />
      )}
      <LeftArrow
        src={resolvePath.image('common/left-arrow.png')}
        data-disabled={page < 2}
        onClick={() => {
          back()
          store.dispatch<any>(
            getConversations(authToken, { term: 'week', page: page - 1 })
          )
        }}
      />
      <RightArrow
        src={resolvePath.image('common/right-arrow.png')}
        onClick={() => {
          forward()
          store.dispatch<any>(
            getConversations(authToken, { term: 'week', page: page + 1 })
          )
        }}
      />
    </Container>
  )
}

const Container = styled.div`
  position: relative;
  width: 100%;
  margin: 0 auto 24px;
  padding-top: 24px;
  background-color: #ffffff;
  ${isHuawei()
    ? 'border: 1px solid rgba(5, 68, 102, 0.3)'
    : 'box-shadow: 0 1px 8px 0 rgba(5, 68, 102, 0.2)'};
`

const LeftArrow = styled.img`
  position: absolute;
  left: 32px;
  bottom: 32px;
  width: 20px;
  height: 34px;

  &[data-disabled='true'] {
    pointer-events: none;
    opacity: 0.4;
  }
`

const RightArrow = styled.img`
  position: absolute;
  right: 32px;
  bottom: 32px;
  width: 20px;
  height: 34px;

  &[data-disabled='true'] {
    pointer-events: none;
    opacity: 0.4;
  }
`
