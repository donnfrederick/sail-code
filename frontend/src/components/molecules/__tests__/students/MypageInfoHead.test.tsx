import MypageInfoHead from 'components/molecules/students/MypageInfoHead'
import * as React from 'react'
import * as renderer from 'react-test-renderer'
import { Intl } from 'components/organisms/Intl'

test('render <MypageInfoHead />', () => {
  const tree = renderer
    .create(
      <Intl>
        <MypageInfoHead
          authToken={''}
          reservations={null}
          forward={() => null}
          back={() => null}
          page={1}
        />
      </Intl>
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
