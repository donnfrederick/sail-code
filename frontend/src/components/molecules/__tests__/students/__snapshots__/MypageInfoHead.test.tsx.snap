// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render <MypageInfoHead /> 1`] = `
<div
  className="sc-gqjmRU dcFyIt"
>
  <img
    className="sc-VigVT gEiuiB"
    data-disabled={true}
    onClick={[Function]}
    src="/assets/img/common/left-arrow.png"
  />
  <img
    className="sc-jTzLTM jwOHkJ"
    onClick={[Function]}
    src="/assets/img/common/right-arrow.png"
  />
</div>
`;
